src/showItems.js

```diff
@@ -0,0 +1,99 @@
+import {
+  listAddTask, updateDescription, listUpdateIndexes, listRemoveTask,
+} from './listManager.js';
+
+export const setTaskListeners = (tasksList, index) => {
+  const text = document.querySelector(`.item-${index} textarea`);
+
+  text.addEventListener('change', () => {
+    updateDescription(tasksList, index, text);
+  });
+
+  text.addEventListener('keypress', (e) => {
+    if (e.key === 'Enter') {
+      e.preventDefault();
+      updateDescription(tasksList, index, text);
+      document.activeElement.blur();
+    }
+  });
+
+  text.addEventListener('focus', () => {
+    const icon = text.nextElementSibling;
+    icon.classList.remove('fa-ellipsis-v');
+    icon.classList.add('fa-trash-alt');
+  });
+
+  text.addEventListener('blur', () => {
+    const icon = text.nextElementSibling;
+    icon.classList.add('fa-ellipsis-v');
+    icon.classList.remove('fa-trash-alt');
+  });
+
+  const check = document.querySelector(`.item-${index} input`);
+  check.addEventListener('change', () => { updateStatus(tasksList, index); });
+
+  const rmBtn = document.querySelector(`.item-${index} i`);
+  // eslint-disable-next-line no-use-before-define
+  rmBtn.addEventListener('click', () => { removeTask(tasksList, index); });
+};
+
+export const showTask = (container, tasksList, index) => {
+  const li = document.createElement('li');
+  li.classList.add('task-item', `item-${index}`);
+  const checkbox = document.createElement('input');
+  checkbox.setAttribute('type', 'checkbox');
+  li.appendChild(checkbox);
+  const text = document.createElement('textarea');
+  text.textContent = tasksList[index].description;
+  li.appendChild(text);
+  const icon = document.createElement('i');
+  icon.classList.add('fas', 'fa-ellipsis-v');
+  li.appendChild(icon);
+  container.appendChild(li);
+  setTaskListeners(tasksList, index);
+};
+
+export const showList = (tasksList) => {
+  const listContainer = document.querySelector('ul');
+  tasksList.forEach((item, index) => {
+    showTask(listContainer, tasksList, index);
+  });
+};
+
+export const updateIndexes = (tasksList) => {
+  listUpdateIndexes(tasksList);
+  const liSet = document.querySelectorAll('.task-item');
+  liSet.forEach((li) => {
+    li.remove();
+  });
+  showList(tasksList);
+};
+
+export const removeTask = (tasksList, index) => {
+  const li = document.querySelector(`.item-${index}`);
+  li.remove();
+  listRemoveTask(tasksList, index);
+  updateIndexes(tasksList);
+};
+
+export const removeMarked = (tasksList) => {
+  let i = 0;
+  while (i < tasksList.length) {
+    if (tasksList[i].completed) {
+      removeTask(tasksList, i);
+    } else {
+      i += 1;
+    }
+  }
+};
+
+export const addTask = (tasksList, addIn) => {
+  if (addIn.value !== '') {
+    const description = addIn.value;
+    listAddTask(tasksList, description);
+    addIn.value = '';
+    const listContainer = document.querySelector('ul');
+    showTask(listContainer, tasksList, tasksList.length - 1);
+  }
```

Comment:
- [ ] Kindly include localStorage functionality when adding and deleting a to-do item. After adding a to-do item and refreshing the page the to-do item should still appear on the page also when the item is deleted it should also be removed from localStorage.

Reply:
All the calls to localStorage for saving the values are made in the module listManager.js, and in this function here is called through the function listAddTask(tasksList, description). At least in my local environment, the functionality works well.